openpype/hosts/maya/plugins/create/create_multishot_layout.py

```diff
@@ -0,0 +1,189 @@
+from ayon_api import (
+    get_folder_by_name,
+    get_folder_by_path,
+    get_folders,
+)
+from maya import cmds  # noqa: F401
+
+from openpype import AYON_SERVER_ENABLED
+from openpype.client import get_assets
+from openpype.hosts.maya.api import plugin
+from openpype.lib import BoolDef, EnumDef, TextDef
+from openpype.pipeline import (
+    Creator,
+    get_current_asset_name,
+    get_current_project_name,
+)
+from openpype.pipeline.create import CreatorError
+
+
+class CreateMultishotLayout(plugin.MayaCreator):
+    """Create a multi-shot layout in the Maya scene.
+
+    This creator will create a Camera Sequencer in the Maya scene based on
+    the shots found under the specified folder. The shots will be added to
+    the sequencer in the order of their clipIn and clipOut values. For each
+    shot a Layout will be created.
+
+    """
+    identifier = "io.openpype.creators.maya.multishotlayout"
+    label = "Multi-shot Layout"
+    family = "layout"
+    icon = "project-diagram"
+
+    def get_pre_create_attr_defs(self):
+        # Present artist with a list of parents of the current context
+        # to choose from. This will be used to get the shots under the
+        # selected folder to create the Camera Sequencer.
+
+        """
+        Todo: get this needs to be switched to get_folder_by_path
+              once the fork to pure AYON is done.
+
+        Warning: this will not work for projects where the asset name
+                 is not unique across the project until the switch mentioned
+                 above is done.
+        """
+
+        current_folder = get_folder_by_name(
+            project_name=get_current_project_name(),
+            folder_name=get_current_asset_name(),
+        )
+
+        current_path_parts = current_folder["path"].split("/")
+
+        items_with_label = [
+            dict(
+                label=current_path_parts[p] if current_path_parts[p] != current_folder["name"] else f"{current_path_parts[p]} (current)",  # noqa
+                value="/".join(current_path_parts[:p + 1]),
+            )
+            for p in range(len(current_path_parts))
+        ]
```

Comment:
Can we rewrite this into something more readable?

Reply:
I've refactored code a little bit and added comments